src/app/content/highlights/reducer.ts

```diff
@@ -80,6 +87,34 @@ const reducer: Reducer<State, AnyAction> = (state = initialState, action) => {
     case getType(actions.clearFocusedHighlight): {
       return omit('focused', state);
     }
+    case getType(actions.setColorsFilter): {
+      const newState = {...state};
+      newState.summary.filters.colors = action.payload;
+      return newState;
+    }
+    case getType(actions.setChaptersFilter): {
+      const newState = {...state};
+      newState.summary.filters.chapters = action.payload;
+      return newState;
+    }
+    case getType(actions.setIsLoadingSummary): {
+      const newState = {...state};
+      newState.summary.loading = action.payload;
+      return newState;
+    }
+    case getType(actions.receiveSummaryHighlights): {
+      const newState = {...state};
+      const { chapters, colors } = newState.summary.filters;
+      if (chapters.length === 0 || colors.length === 0) {
+        // When we make api call without filters it is returning all highlights
+        // so we manually set it to empty object.
```

Comment:
the reducer shouldn't care about this, the hook should detect this case and short circuit before actually loading all highlights